MC-35509: Updated the software update guide module with the safe upgrade tool documentation

danidelcar opened the pull request:
## Purpose of this pull request

This pull request (PR) adds Safe Upgrade Tool (SUT) topics to the Software Update Guide section in version 2.4.

## Affected DevDocs pages

<!-- REQUIRED List the affected pages on devdocs.magento.com (URLs). Not needed for large numbers of files. -->

This PR includes the following pages:

- [SUT welcome page](https://devdocs.magento.com/safe-upgrade-tool/introduction.html)
- [Prerequisites for SUT](https://devdocs.magento.com/safe-upgrade-tool/prerequisites.html)
- [SUT installation page](https://devdocs.magento.com/safe-upgrade-tool/install.html)
- [SUT use page](https://devdocs.magento.com/safe-upgrade-tool/run.html)
- [SUT developer information page](https://devdocs.magento.com/safe-upgrade-tool/developer.html)
- [SUT tracking page](https://devdocs.magento.com/safe-upgrade-tool/tracking.html)

This PR modifies the following pages:

- [v2.4 Software Update Guide](https://devdocs.magento.com/guides/v2.4/comp-mgr/bk-compman-upgrade-guide.html)
- [v2.3 Software Update Guide](https://devdocs.magento.com/guides/v2.3/comp-mgr/bk-compman-upgrade-guide.html)

## Other information

JIRA [MC-35509](https://jira.corp.magento.com/browse/MC-35509)

whatsnew
Added the [Safe Upgrade Tool](https://devdocs.magento.com/safe-upgrade-tool/introduction.html) topics to the Software Update Guide section.

danidelcar commented:
closing it due to CLA sign, will reopen to run sign process again

jeff-matthews commented:
@danidelcar, let me know when you're ready for review. I went ahead and assigned myself to prevent anyone else from self-assigning, but I won't start my review until your tell me you're ready.

danidelcar commented:
> @danidelcar, let me know when you're ready for review. I went ahead and assigned myself to prevent anyone else from self-assigning, but I won't start my review until your tell me you're ready.

thanks @jeff-matthews !, once I solve the Adobe CLA, I can let you know. The PR is ready for review in any case

danidelcar commented:
Hi @adobe/open-source-office, I am having issues signing the Adobe CLA, but I did request access months ago, and I did it again, but still is failing to sign the Adobe CLA properly.

jeff-matthews commented on src/guides/sut/sut-tracking.md:
```diff
@@ -0,0 +1,49 @@
+
+You can delete the SUT tracking by executing the following command:
+
+`aws cloudformation delete-stack --stack-name sut-tracking`
```
Wrap in code fences

danidelcar replied:
done

jeff-matthews commented on src/guides/sut/sut-welcome-guide.md:
```diff
@@ -0,0 +1,88 @@
+  - Upgrade
+---
+
+Welcome to the Magento Safe Upgrade Tool (SUT) page.
```
```suggestion

```

jeff-matthews reviewed (changes requested):
This is a good start, but I have several editorial comments and suggestions. 

I couldn't finish the review, but I wanted to submit what I have so far. I will need to do another review pass later.

In addition to the inline comments and suggestions, I also need you to do the following:

- Move the `sut/` directory from `devdocs/src/guides/sut/` to `devdocs/src/sut/`.
- Change the name of the `sut/` directory to `safe-upgrade-tool/`.
- For beta, the `src/guides/sut/img/mvp-diagram.png` diagram is fine, but for release, I'd like to see a more refined diagram/infographic that uses our doc design standards for graphics.

danidelcar commented on src/guides/sut/prereq-sut-checklist.md:
```diff
@@ -0,0 +1,24 @@
+{:style="table-layout:auto;"}
+
+{:.bs-callout .bs-callout-info}
+SUT should run in any operating system. It is not required to run the tool where your magento instance is located.
```
Yes, and if I'm not mistaken, running sut in the same location could present some issues, as per the workshops we had last weeks

danidelcar commented on src/guides/sut/sut-dev-guide.md:
```diff
@@ -0,0 +1,130 @@
+
+## M-Ray integration
+
+The integration with M-Ray is done through:
```
I'll look into that, thanks Jeff!

danidelcar commented on src/guides/sut/sut-dev-guide.md:
```diff
@@ -0,0 +1,130 @@
+
+`Sut\Domain\MRay\MRayInterface : "@sut_mray_mock"`
+
+In this example, SUT uses `MRayMock` as the `MRayInterface` implementation. Then, the responses of the methods `api()` and `modules()` come from:
```
yes, maybe it is clearer that way, thanks!

danidelcar commented on src/guides/sut/sut-dev-guide.md:
```diff
@@ -0,0 +1,130 @@
+*  `dev/mray_mock_files/api.json`
+*  `dev/mray_mock_files/modules.json`
+
+{:.bs-callout .bs-callout-info}
```
in oms we use the callout-info as I included it here, should we change that in oms too?

danidelcar commented on src/guides/sut/sut-dev-guide.md:
```diff
@@ -0,0 +1,130 @@
+
+## Unit testing
+
+To run the unit tests, you could execute one of the following commands:
```
I do not see a difference, the command is the same to run all tests, they change some of the variable, but in essence is the same

danidelcar commented on src/guides/sut/sut-dev-guide.md:
```diff
@@ -0,0 +1,130 @@
+
+This file can be edited to customize the response according to your installation. Just replace the value assigned to `Sut\Domain\MRay\MRayInterface`:
+
+> Example of a custom value
```
we did use those in OMS, I'll include it as a H3, there are several of this type of introduction throughout these sut docs

danidelcar replied:
modified them

danidelcar commented on src/guides/sut/sut-dev-guide.md:
```diff
@@ -0,0 +1,130 @@
+{:.bs-callout .bs-callout-info}
+When you make changes in `services.yaml`, delete the cache folder (`var/cache`) to correctly apply them.
+
+See the [magento extensions](https://docs.magento.com/user-guide/magento/magento-extensions.html) topic for more information on extensions developed and released by Magento.
```
due to this m-ray integration. Seems it is a library that analyses extensions and magento code, I must have put it as a anchor point for a link to m-ray, but still trying to find out about it from devs.

danidelcar commented on src/guides/sut/sut-dev-guide.md:
```diff
@@ -0,0 +1,130 @@
+
+In order to distribute SUT:
+
+1. Go to the [Magento create repo](https://repo.magento.com/admin/packagist/web/ceeerelease/create) topic and create a new release with the following parameters:
```
same here, I will chase that down

danidelcar commented on src/guides/sut/sut-dev-guide.md:
```diff
@@ -0,0 +1,130 @@
+
+   *  Name: `magento/safe-upgrade-tool-VERSION`
+
+1. Go to the [Magento upload repo](https://repo.magento.com/admin/upload_m2_version) topic to upload the `sut.zip` with the following parameters:
```
same as previous comment

danidelcar commented on src/guides/sut/sut-dev-guide.md:
```diff
@@ -0,0 +1,130 @@
+   *  Release: `magento/safe-upgrade-tool-VERSION`
+   *  Edition: CE
+
+1. Finally, go to the [Magento version repo](https://repo.magento.com/admin/packagist/web/version/list) topic to check that the new version has been created.
```
same as previous comment

danidelcar commented on src/guides/sut/sut-dev-guide.md:
```diff
@@ -0,0 +1,130 @@
+``Complexity Score = 2 * (# of errors) + 1 * (# of warnings)``
+
+## SUT packaging
+
```
expanded information in this section

danidelcar commented on src/guides/sut/sut-dev-guide.md:
```diff
@@ -0,0 +1,130 @@
+1. This generates a `sut.zip` file that we it is uploaded to the [Magento repo page](https://repo.magento.com).
+
+## SUT distribution
+
```
expanded information in this section

danidelcar commented on src/guides/sut/sut-dev-guide.md:
```diff
@@ -0,0 +1,130 @@
+
+## SUT tracking
+
+SUT tracking is a small AWS lambda done with GO. The code is available in the the [SUT github repository](https://github.com/magento-commerce/safe-upgrade-tool), inside the `sut-tracking` folder.
```
included correct url

danidelcar commented on src/guides/sut/sut-tracking.md:
```diff
@@ -0,0 +1,49 @@
+  - Upgrade
+---
+
+SUT tracking is a small AWS lambda done with GO. The code is available in the [SUT github repository](https://github.com/magento-commerce/safe-upgrade-tool), inside the `sut-tracking` folder.
```
included correct url

danidelcar commented on src/guides/sut/sut-tracking.md:
```diff
@@ -0,0 +1,49 @@
+
+## Requirements
+
+In order to install the requirements for the SUT tracking execute the following command:
```
modified it

danidelcar commented on src/guides/sut/sut-tracking.md:
```diff
@@ -0,0 +1,49 @@
+
+Then, run:
+
+`curl "https://awscli.amazonaws.com/AWSCLIV2.pkg" -o "AWSCLIV2.pkg" sudo installer -pkg AWSCLIV2.pkg -target /`
```
modified it

danidelcar commented on src/guides/sut/sut-tracking.md:
```diff
@@ -0,0 +1,49 @@
+
+In order to deploy the SUT tracking:
+
+1. Go to [KLAM](https://klam.corp.adobe.com/)
```
what should we do with a url like this one?, could we discuss this with Sandra later today?

jeff-matthews replied:
Is this a workflow we expect external devs to use or is this just for internal devs? It relates to the questions that I have about Packaging and Distributing to `repo.magento.com` in the developer guide. As far as I know, only internal devs have access to upload packages to our official Composer repo (`repo.magento.com`).

danidelcar commented on src/guides/sut/sut-tracking.md:
```diff
@@ -0,0 +1,49 @@
+1. Then, edit the `template.yaml` file and set the `NEW_RELIC_API_KEY` with a valid newrelic api key
+1. Finally, deploy SUT tracking by executing one of these commands:
+
+  `sam deploy --guided`
```
done

danidelcar commented on src/guides/sut/sut-tracking.md:
```diff
@@ -0,0 +1,49 @@
+1. Finally, deploy SUT tracking by executing one of these commands:
+
+  `sam deploy --guided`
+  `sam deploy`
```
done

jeff-matthews commented:
@danidelcar, since you're out this week, I took the liberty of pushing a few changes directly to your branch. When you get back, be sure to pull changes before making any local changes.

You'll still need to get answers to some of the outstanding workflow questions re: packaging, distributing, and deploying tracking, but I'll generate a staging preview and send it to Sandra in the meantime.

danidelcar commented:
> @danidelcar, since you're out this week, I took the liberty of pushing a few changes directly to your branch. When you get back, be sure to pull changes before making any local changes.
> 
> You'll still need to get answers to some of the outstanding workflow questions re: packaging, distributing, and deploying tracking, but I'll generate a staging preview and send it to Sandra in the meantime.

Thanks @jeff-matthews I pulled changes, meanwhile I will clarify the doubts we had!

danidelcar commented:
@jeff-matthews clarified tracking topic, better to have it only for internal use, so I will remove the file from the PR

danidelcar commented:
Already working on some changes for distribution

danidelcar commented:
Hi @jeff-matthews & @leslietilling I modified several things:

- Corrected whatsnew entry
- Added entry in page-header.html for the banner
- Corrected pages
- Deleted tracking topic

jeff-matthews commented on src/_includes/layout/page-header.html:
```diff
@@ -42,4 +42,9 @@ <h2 class="page-subtitle no_toc">{{ page.subtitle }}</h2>
 
+  {% if page.url contains "safe-upgrade-tool/" %}
+  <div class="message-banner">
+    This tool is still in ALPHA and is not yet available for all customers. Submit a Magento Support ticket for assistance and questions.  </div>
```
Does Magento Support know that we're going to direct people to them for SUT questions?

danidelcar replied:
they normally contact support or submit a ticket to request an account for magento composer if I'm not mistaken, no?

danidelcar replied:
Should I discuss this with someone to get a proper clarification about that?, in support?, or Sandra to see if she thought about that?. This comes after the conversations I had with ENG side to clarify magento composer

jeff-matthews replied:
@danidelcar, please confirm messaging with Sandra.

People usually self-serve on creating a Magento account to get access keys for repo.magento.com. The only time they contact Magento Support is when they're keys don't work as expected or they don't have the proper entitlements/permissions to packages on repo.magento.com.

sgonzalezmangana replied:
In January we will open the tool and it will be available for all Commerce merchants, so we can change this phrase for: "This tool is still in **ALPHA** with limited scope. If you are a Commerce user you can download it from repo.magento.com, we are looking forward to get your feedback."

danidelcar replied:
corrected include

jeff-matthews commented on src/guides/v2.3/comp-mgr/bk-compman-upgrade-guide.md:
```diff
@@ -53,6 +53,8 @@ Related topics
 Complete the tasks discussed in [Prerequisites].
 
+ {:.bs-callout-info}
+See the [Safe upgrade tool section]({{ site.baseurl }}/safe-upgrade-tool/introduction.html) for more information on the new CLI magento tool that helps you update your magento software.
```
```suggestion
See the [Safe Upgrade Tool ]({{ site.baseurl }}/safe-upgrade-tool/introduction.html) for more information about the new Magento CLI tool that helps you update your Magento software.
```

sgonzalezmangana replied:
Could we add "(ALPHA)" next to Safe Upgrade Tool ?

danidelcar replied:
added

jeff-matthews commented on src/guides/v2.4/comp-mgr/bk-compman-upgrade-guide.md:
```diff
@@ -48,6 +48,9 @@ Related topics
 Complete the tasks discussed in [Prerequisites].
 
+ {:.bs-callout-info}
+See the [Safe upgrade tool section]({{ site.baseurl }}/safe-upgrade-tool/introduction.html) for more information on the new CLI magento tool that helps you update your magento software.
```
```suggestion
See the [Safe Upgrade Tool]({{ site.baseurl }}/safe-upgrade-tool/introduction.html) for more information about the new Magento CLI tool that helps you update your Magento software.
```

sgonzalezmangana replied:
Same as before, Could we add "(ALPHA)" next to Safe Upgrade Tool?

jeff-matthews commented on src/safe-upgrade-tool/developer.md:
```diff
@@ -0,0 +1,150 @@
+
+## M-Ray integration
+
+M-Ray integration is an internal integration solution.
```
I'm still not sure what M-Ray is. If it's internal, will the public have access to it?

danidelcar replied:
I'll chase that down, but I think it is an internal developed solution, but devs can access it

paliarush replied:
Initially M-Ray along with SUT was supposed to be available only to Commerce customers. Has this changed and approved by product organization?

sgonzalezmangana replied:
@paliarush SUT will be limited to Commerce merchants as you pointed out there is no change on this. 
I think we should talk more about Magento API index instead of M-Ray, as M-Ray is an internal tool that won't be used by SUT users but the output (Magento API index) of it will. @paliarush what do you think?

danidelcar replied:
for the moment, changed it to Magento API index

jeff-matthews commented on src/safe-upgrade-tool/developer.md:
```diff
@@ -0,0 +1,150 @@
+
+`Sut\Domain\MRay\MRayInterface`
+
+Its implementation is through the `config/services.yaml` file. Its value decides where the response of methods `api()` and `modules()` comes from.
```
```suggestion
It is implemented through the `config/services.yaml` file. Its value decides where the response of methods `api()` and `modules()` comes from.
```

jeff-matthews commented on src/safe-upgrade-tool/developer.md:
```diff
@@ -0,0 +1,150 @@
+1. Change the `TESTS_BASE_URL` value to point to the Magento URL that you want to test.
+1. To run the acceptance tests, execute one of the following commands:
+
+*  `vendor/bin/phpunit -c tests/acceptance/phpunit.xml tests/acceptance`
```
This unordered list needs to be indented so that it aligns under the preceding ordered list.

jeff-matthews commented on src/safe-upgrade-tool/developer.md:
```diff
@@ -0,0 +1,150 @@
+
+A package is essentially a directory containing something. It contains a package description which has a name and a version. The name and the version are used to identify the package.
+
+It is possible to create a `sut.zip` file of the SUT to distribute it with partners as a composer package.
```
```suggestion
It is possible to create a `sut.zip` file of the SUT to distribute it as a Composer package.
```

jeff-matthews commented on src/safe-upgrade-tool/developer.md:
```diff
@@ -0,0 +1,150 @@
+
+`Complexity Score = 2 * (# of errors) + 1 * (# of warnings)`
+
+## Packaging
```
Have you confirmed whether external users can actually do packaging? This seems like an internal-only task.

paliarush replied:
Totally agree, there is no reason to have this section in public docs.

danidelcar replied:
will remove all packaging section then

jeff-matthews commented on src/safe-upgrade-tool/developer.md:
```diff
@@ -0,0 +1,150 @@
+
+## Packaging
+
+A package is essentially a directory containing something. It contains a package description which has a name and a version. The name and the version are used to identify the package.
```
This paragraph is awkward. It sounds like we're referring to a Composer package or Composer metapackage. 

https://getcomposer.org/doc/04-schema.md#type

If so, please use one of our official descriptions in the Magento Glossary:

- https://glossary.magento.com/composer-package/
- https://glossary.magento.com/metapackage/

danidelcar replied:
will remove all packaging section then

jeff-matthews commented on src/safe-upgrade-tool/developer.md:
```diff
@@ -0,0 +1,150 @@
+
+It is possible to create a `sut.zip` file of the SUT to distribute it with partners as a composer package.
+
+In order to upload it to be able to distribute it, you need first to create the package:
```
```suggestion
```

jeff-matthews commented on src/safe-upgrade-tool/developer.md:
```diff
@@ -0,0 +1,150 @@
+
+In order to upload it to be able to distribute it, you need first to create the package:
+
+1. To package the SUT execute the following command:
```
```suggestion
1. To package the SUT, execute the following command:
```

jeff-matthews commented on src/safe-upgrade-tool/developer.md:
```diff
@@ -0,0 +1,150 @@
+
+`bin/package 0.0.1`
+
+1. This generates an `sut.zip` file that you can upload (or distribute with) to the [Magento Composer repository](https://repo.magento.com).
```
```suggestion
1. This generates an `sut.zip` file that you can upload to the [Magento Composer repository](https://repo.magento.com).
```

jeff-matthews replied:
As far as I know, only authorized employees can upload packages to repo.magento.com. Are you sure this belongs in our public docs?

danidelcar replied:
will remove all packaging section then

jeff-matthews commented on src/safe-upgrade-tool/developer.md:
```diff
@@ -0,0 +1,150 @@
+{:.bs-callout-info}
+A repository is a package source. Composer will look in all your repositories to find the packages your SUT requires.
+
+If you need access to the [Magento Composer repository](https://repo.magento.com), submit a [Magento Support ticket]({{site.baseurl}}/cloud/trouble/trouble.html) to request permissions and an account.
```
Does support know we're directing users to them for assistance with SUT and repo.magento.com access?

danidelcar replied:
will remove all packaging section then

jeff-matthews commented on src/safe-upgrade-tool/developer.md:
```diff
@@ -0,0 +1,150 @@
+
+In order to distribute SUT:
+
+1. Go to the [Magento create repo](https://repo.magento.com/admin/packagist/web/ceeerelease/create) topic and create a new release with the following parameters:
```
```suggestion
1. Go to the [Magento create repo](https://repo.magento.com/admin/packagist/web/ceeerelease/create) page and create a new release with the following parameters:
```

jeff-matthews commented on src/safe-upgrade-tool/developer.md:
```diff
@@ -0,0 +1,150 @@
+
+   *  Name: `magento/safe-upgrade-tool-VERSION`
+
+1. Go to the [Magento upload repo](https://repo.magento.com/admin/upload_m2_version) topic to upload the `sut.zip` with the following parameters:
```
```suggestion
1. Go to the [Magento upload repo](https://repo.magento.com/admin/upload_m2_version) page to upload the `sut.zip` file with the following parameters:
```

jeff-matthews commented on src/safe-upgrade-tool/developer.md:
```diff
@@ -0,0 +1,150 @@
+   *  Release: `magento/safe-upgrade-tool-VERSION`
+   *  Edition: CE
+
+1. Finally, go to the [Magento version repo](https://repo.magento.com/admin/packagist/web/version/list) topic to check that the new version has been created.
```
```suggestion
1. Finally, go to the [Magento version repo](https://repo.magento.com/admin/packagist/web/version/list) page to check that the new version has been created.
```

sgonzalezmangana replied:
We need to remove all the package section as highlighted by @paliarush

danidelcar replied:
will remove all packaging section then

danidelcar replied:
what about distribution @sgonzalezmangana?

danidelcar replied:
removed it, as per other comments

jeff-matthews commented on src/safe-upgrade-tool/install.md:
```diff
@@ -0,0 +1,80 @@
+
+The following use case describes the typical process for a Magento partner to upgrade a client's Magento instance:
+
+1. A partner's software Engineer downloads the SUT and executes it during the beta phase of the newest Magento release.
```
```suggestion
1. A partner's Software Engineer downloads the SUT and executes it during the beta phase of the newest Magento release.
```

jeff-matthews commented on src/safe-upgrade-tool/install.md:
```diff
@@ -0,0 +1,80 @@
+The following use case describes the typical process for a Magento partner to upgrade a client's Magento instance:
+
+1. A partner's software Engineer downloads the SUT and executes it during the beta phase of the newest Magento release.
+1. The Engineer sees that there are several customized areas broken in the inventory and catalog modules and they also get a complexity score of X. See the [Developer information guide]({{ site.baseurl }}/safe-upgrade-tool/developer.html) for more information on the complexity score.
```
```suggestion
1. The Software Engineer sees that there are several customized areas broken in the inventory and catalog modules and they also get a complexity score of X. See the [Developer information guide]({{ site.baseurl }}/safe-upgrade-tool/developer.html) for more information on the complexity score.
```

jeff-matthews commented on src/safe-upgrade-tool/install.md:
```diff
@@ -0,0 +1,80 @@
+
+1. A partner's software Engineer downloads the SUT and executes it during the beta phase of the newest Magento release.
+1. The Engineer sees that there are several customized areas broken in the inventory and catalog modules and they also get a complexity score of X. See the [Developer information guide]({{ site.baseurl }}/safe-upgrade-tool/developer.html) for more information on the complexity score.
+1. With this information, the Engineer is able to understand the complexity of the upgrade and is able to relay this information back to the partner's Account Manager.
```
```suggestion
1. With this information, the Software Engineer is able to understand the complexity of the upgrade and is able to relay this information back to the partner's Account Manager.
```

jeff-matthews commented on src/safe-upgrade-tool/install.md:
```diff
@@ -0,0 +1,80 @@
+1. The Engineer sees that there are several customized areas broken in the inventory and catalog modules and they also get a complexity score of X. See the [Developer information guide]({{ site.baseurl }}/safe-upgrade-tool/developer.html) for more information on the complexity score.
+1. With this information, the Engineer is able to understand the complexity of the upgrade and is able to relay this information back to the partner's Account Manager.
+1. The Account Manager creates a timeline and cost for the Magento upgrade, which allows them to get their manager's approval.
+1. With their manager's approval, the Engineer works on the required code modifications to fix the broken modules.
```
```suggestion
1. With their manager's approval, the Software Engineer works on the required code modifications to fix the broken modules.
```

jeff-matthews commented on src/safe-upgrade-tool/install.md:
```diff
@@ -0,0 +1,80 @@
+1. With this information, the Engineer is able to understand the complexity of the upgrade and is able to relay this information back to the partner's Account Manager.
+1. The Account Manager creates a timeline and cost for the Magento upgrade, which allows them to get their manager's approval.
+1. With their manager's approval, the Engineer works on the required code modifications to fix the broken modules.
+1. The Engineer runs the SUT tool one more time with a Magento pre-release to ensure there are no new issues and that their code changes fixed the problems found during the beta phase.
```
```suggestion
1. The Software Engineer runs the SUT tool one more time with a Magento pre-release to ensure there are no new issues and that their code changes fixed the problems found during the beta phase.
```

jeff-matthews commented on src/safe-upgrade-tool/install.md:
```diff
@@ -0,0 +1,80 @@
+1. The Account Manager creates a timeline and cost for the Magento upgrade, which allows them to get their manager's approval.
+1. With their manager's approval, the Engineer works on the required code modifications to fix the broken modules.
+1. The Engineer runs the SUT tool one more time with a Magento pre-release to ensure there are no new issues and that their code changes fixed the problems found during the beta phase.
+1. Everything check’s out and the Engineeer pushes the code to a staging environment where automated tests confirm all tests are green, which allows them to release the latest Magento version to production the same day that the Magento pre-release is released.
```
```suggestion
1. Everything check’s out and the Software Engineer pushes the code to a staging environment where automated tests confirm all tests are green, which allows them to release the latest Magento version to production the same day that the Magento pre-release is released.
```

paliarush replied:
The reality is almost no system integrator has automated tests. Again, this is desired state. 

It would be more accurate to mention "regression testing" phase without specifying whether it is completed manually or automatically.

jeff-matthews commented on src/safe-upgrade-tool/introduction.md:
```diff
@@ -0,0 +1,16 @@
+
+The tool returns a list of errors and warnings that you must address before upgrading to a new version of Magento.
+
+This tool is distributed as a composer package with every release of a Magento version. See [Developer]({{site.baseurl}}/safe-upgrade-tool/developer.html) topic for more information.
```
```suggestion
This tool is distributed as a Composer package with every release of a Magento version. See [Developer]({{site.baseurl}}/safe-upgrade-tool/developer.html) topic for more information.
```

jeff-matthews commented on src/safe-upgrade-tool/run.md:
```diff
@@ -0,0 +1,137 @@
+
+The Magento Safe Upgrade Tool (SUT) is a command line (CLI) tool that checks a Magento instance against a specific version by analyzing all the non-Magento modules installed on it.
+
+potential problems that must be fixed in your custom code before attempting to upgrade to a newer version of Magento.
```
```suggestion
The SUT identifies potential problems that must be fixed in your custom code before attempting to upgrade to a newer version of Magento.
```

sgonzalezmangana replied:
Let's add "(ALPHA)" next to Safe Upgrade Tool.
After this paragraph, we should also ad an explanation about Alpha and its limited scope.

"At the moment this is an Alpha version with limited scope, only validating PHP Magento APIs and GraphQL schema!

danidelcar replied:
added

paliarush commented on src/safe-upgrade-tool/developer.md:
```diff
@@ -0,0 +1,150 @@
+
+## M-Ray integration
+
+M-Ray integration is an internal integration solution.
```
```suggestion
Replace with brief overview of M-Ray and its purpose.
```

paliarush commented on src/safe-upgrade-tool/developer.md:
```diff
@@ -0,0 +1,150 @@
+
+### Complexity score formula
+
+`Complexity Score = 2 * (# of errors) + 1 * (# of warnings)`
```
Above, the score is said to be in range (0, 100), which is normalized value.
While this formula is for absolute values, which can go beyond 100.

Please clarify with the team which one is currently implemented. Originally it was absolute value, and I have not seen it to be changed to normalized one yet.

sgonzalezmangana replied:
Yes, it is an absolute value, we are planning to normalize it but it is still not implemented

danidelcar replied:
so how we could reflect this in the docs then?, should we change the absolute values mentioned in the formula, or modify the range, which is a normalized value?

danidelcar replied:
depending on when are we applying these changes

danidelcar replied:
@svera could you take a look at this thread too?, thanks!

svera replied:
@danidelcar right now the range can be from 0 to ∞. As Sandra said, we plan to implement a formula to return normalized values, but is not defined yet.

danidelcar replied:
corrected section, please check @paliarush & @svera

paliarush commented on src/safe-upgrade-tool/developer.md:
```diff
@@ -0,0 +1,150 @@
+
+For more information about Composer, see the [Introduction to Magento Composer]({{site.baseurl}}/guides/v2.4/extension-dev-guide/intro/intro-composer.html) topic.
+
+## Distribution
```
This section also does not make sense for external developers.

paliarush commented on src/safe-upgrade-tool/install.md:
```diff
@@ -0,0 +1,80 @@
+
+### Who is the SUT for?
+
+The following use case describes the typical process for a Magento partner to upgrade a client's Magento instance:
```
The described scenario is our "desired state", not a reality with current version of SUT.

At the moment SUT is capable of analyzing only a fraction of all extensibility scenarios. It means that system integrators must not exclusively rely on SUT output to forecast timeline and cost of the upgrade.

This may be important, depending on our goal with this documentation.

danidelcar replied:
I would understand that being an ALPHA, that section would be ideal scenario, which we can still keep, if @sgonzalezmangana thinks worth it, we could add a clarification for it

paliarush commented on src/safe-upgrade-tool/install.md:
```diff
@@ -0,0 +1,80 @@
+See [prerequisites]({{ site.baseurl }}/safe-upgrade-tool/prerequisites.html).
+
+{:.bs-callout-info}
+You can run the SUT in any operating system. There is no requirement to run the SUT where your Magento instance is located. For example, you can install the SUT on one server and point it at your Magento installation on another server.
```
It is necessary for SUT to have access to the source code of the Magento instance.

paliarush commented on src/safe-upgrade-tool/install.md:
```diff
@@ -0,0 +1,80 @@
+
+*  Magento access keys
+*  Composer
+*  Node.js
```
Node JS is only a requirement if they wish to check GraphQL compatibility, otherwise it is not needed.

paliarush commented on src/safe-upgrade-tool/install.md:
```diff
@@ -0,0 +1,80 @@
+
+## Third-party extensions
+
+Magento recommends that you contact your search engine vendor to determine whether your extension is fully compatible with Magento 2.4.
```
Not sure why we have this section about "contacting search engine vendor", please clarify.

danidelcar replied:
As I understood it, contact your 3rd party vendor to check the extension is compatible with 2.4?.

sgonzalezmangana replied:
We should remove the "Third-party extensions" content as it is not relevant at this point for SUT

paliarush commented on src/safe-upgrade-tool/introduction.md:
```diff
@@ -0,0 +1,16 @@
+
+The Safe Upgrade Tool (SUT) is a command line (CLI) tool that checks a Magento instance against a specific version by analyzing all the non-Magento modules installed in it.
+
+The SUT identifies potential problems that must be fixed in your custom code before attempting to upgrade to a newer version of Magento.
```
The tool reports Warnings and Errors. Only errors indicate some broken scenarios.

sgonzalezmangana replied:
Let's add "(ALPHA)" next to Safe Upgrade Tool.

After this paragraph, we should also add an explanation about Alpha and its limited scope.

"At the moment this is an Alpha version with limited scope, only validating PHP Magento APIs and GraphQL schema!

paliarush commented on src/safe-upgrade-tool/introduction.md:
```diff
@@ -0,0 +1,16 @@
+
+The tool returns a list of errors and warnings that you must address before upgrading to a new version of Magento.
+
+This tool is distributed as a composer package with every release of a Magento version. See [Developer]({{site.baseurl}}/safe-upgrade-tool/developer.html) topic for more information.
```
As far as I know it has not been decided yet that we want to package the tool with EACH release of Magento, including all beta versions. This needs to be clarified with Sandra

sgonzalezmangana replied:
We should have a version of the tool for each released version. @alena-marchenko will be working on an automated process to generate the Magento API index with every new version we release.

danidelcar replied:
should we keep this paragraph as is then?

paliarush commented on src/safe-upgrade-tool/prerequisites.md:
```diff
@@ -0,0 +1,22 @@
+|----------------|-----------------|
+| PHP version| >= 7.3 |
+| Composer | none |
+| Node.js | none |
```
Not JS is optional, and is only needed for verifying GraphQL backward compatibility testing.

paliarush commented on src/safe-upgrade-tool/prerequisites.md:
```diff
@@ -0,0 +1,22 @@
+{:style="table-layout:auto;"}
+
+{:.bs-callout-info}
+You can run the SUT in any operating system. There is no requirement to run the SUT where your Magento instance is located. For example, you can install SUT on one server and point it at your Magento installation on another server. Refer to the [install]({{site.baseurl}}/safe-upgrade-tool/install.html#install).
```
There is identical section in  another file. 
SUT must have access to Magento instance source code.

danidelcar replied:
Added info

paliarush commented on src/safe-upgrade-tool/run.md:
```diff
@@ -0,0 +1,137 @@
+
+### Memory limitations
+
+The SUT uses memory on your Magento instance to run—even if you installed the SUT on a different machine. If you are running a system with large third-party modules and files, the SUT requires a high amount of RAM.
```
We should give estimate of what "high amount of RAM" is.

danidelcar replied:
Do we have info regarding this estimate @sgonzalezmangana or @svera ?

svera replied:
@danidelcar it greatly depends on the size of the scanned instance, but a fair assumption is at least 2 GB RAM. Of course, the more the better.

danidelcar replied:
thanks @svera

danidelcar replied:
Added info

paliarush commented on src/safe-upgrade-tool/run.md:
```diff
@@ -0,0 +1,137 @@
+
+### Memory limitations
+
+The SUT uses memory on your Magento instance to run—even if you installed the SUT on a different machine. If you are running a system with large third-party modules and files, the SUT requires a high amount of RAM.
```
```suggestion
The SUT uses memory on your Magento instance to run - even if you installed the SUT on a different machine. If you are running a system with large third-party modules and files, the SUT requires a high amount of RAM.
```

paliarush commented on src/safe-upgrade-tool/run.md:
```diff
@@ -0,0 +1,137 @@
+
+### Memory limitations
+
+The SUT uses memory on your Magento instance to run—even if you installed the SUT on a different machine. If you are running a system with large third-party modules and files, the SUT requires a high amount of RAM.
```
Please confirm with the team that this statement is true, sounds strange to me:
`The SUT uses memory on your Magento instance to run - even if you installed the SUT on a different machine`

danidelcar replied:
@svera could you help clarify this section?

svera replied:
@danidelcar I would get rid of the whole phrase, it doesn't make sense.

danidelcar replied:
The whole memory limitations section @svera ?, or just the phrase Alex mentioned earlier:

`The SUT uses memory on your Magento instance to run — even if you installed the SUT on a different machine. `

We do need to highlight the memory limitations we face with the tool I think

svera replied:
@del just the phrase Alex mentioned, the memory requirements are important.

danidelcar replied:
corrected section, please review @svera & @paliarush  thanks!

paliarush commented on src/safe-upgrade-tool/run.md:
```diff
@@ -0,0 +1,137 @@
+
+### GraphQL schema compatibility verification
+
+The SUT also provides the option to introspect two GraphQL endpoints and compare their schemas looking for breaking and dangerous changes between them:
```
It is important to mention, that in this case they must have **running** "instance before" and "instance after" the upgrade.

paliarush reviewed (changes requested):
1. If our goal is to bring the value to customers, it may make sense to reference (promote) https://github.com/AmpersandHQ/ampersand-magento2-upgrade-patch-helper, and mention that SUT complements Ampersand's tool, not replaces. Please clarify with Sandra.
2. We were planning to prepare docker file to simplify installation steps. Please check with Sandra if this is still planned for the near future, and if so - this can help to simplify docs, improve dev experience and increase adoption

sgonzalezmangana commented on src/safe-upgrade-tool/developer.md:
```diff
@@ -0,0 +1,150 @@
+  - Upgrade
+---
+
+This topic contains information for developers who want to know more technical information about the Magento Safe Upgrade Tool (SUT). It is focused on developers who work closely with the Magento source code. You can use this knowledge to customize the SUT's components.
```
Could we add "(ALPHA)" next to Safe Upgrade Tool ?

sgonzalezmangana commented on src/safe-upgrade-tool/install.md:
```diff
@@ -0,0 +1,80 @@
+  - Upgrade
+---
+
+The Safe Upgrade Tool (SUT) is a command line (CLI) tool that checks a Magento instance against a specific version by analyzing all the non-Magento modules installed on it.
```
Could we add "(ALPHA)" next to Safe Upgrade Tool?

danidelcar commented:
> 1. If our goal is to bring the value to customers, it may make sense to reference (promote) https://github.com/AmpersandHQ/ampersand-magento2-upgrade-patch-helper, and mention that SUT complements Ampersand's tool, not replaces. Please clarify with Sandra.

makes sense @paliarush , I will add this info as a link (reference) in the introduction topic. @sgonzalezmangana what do you think?

danidelcar commented:
> 1. If our goal is to bring the value to customers, it may make sense to reference (promote) https://github.com/AmpersandHQ/ampersand-magento2-upgrade-patch-helper, and mention that SUT complements Ampersand's tool, not replaces. Please clarify with Sandra.
> 2. We were planning to prepare docker file to simplify installation steps. Please check with Sandra if this is still planned for the near future, and if so - this can help to simplify docs, improve dev experience and increase adoption

As per the docker file, do we have any information right now to be added to the docs PR @sgonzalezmangana ?

jeff-matthews reviewed (approved):
Editorial review approved!

It looks like you're still working on the technical review before merging, which is fine.

hostep commented on src/safe-upgrade-tool/introduction.md:
```diff
@@ -0,0 +1,19 @@
+
+The Safe Upgrade Tool ALPHA (SUT) is a command line (CLI) tool that checks a Magento instance against a specific version by analyzing all the non-Magento modules installed in it.
+
+The SUT identifies potential problems that must be fixed in your custom code before attempting to upgrade to a newer version of Magento. SUT complements another tool called [Ampersand]((https://github.com/AmpersandHQ/ampersand-magento2-upgrade-patch-helper).
```
["Ampersand"](https://ampersandcommerce.com/) is the company behind this tool, the tool itself is called "ampersand-magento2-upgrade-patch-helper". So you'd best change that. Also it looks like there is a `(` too much here.

danidelcar replied:
Thanks for the correction @hostep !, much appreciated

paliarush commented on src/safe-upgrade-tool/run.md:
```diff
@@ -0,0 +1,142 @@
+
+### Memory limitations
+
+If you are running a system with large third-party modules and files, the SUT requires a high amount of RAM, at least 2GB RAM.
```
Looks good now

danidelcar commented:
running tests

danidelcar commented:
@sgonzalezmangana can you take a final look at the PR, in case we want to merge tomorrow after the demo?, thanks!

sgonzalezmangana commented on src/safe-upgrade-tool/introduction.md:
```diff
@@ -0,0 +1,20 @@
+
+The Safe Upgrade Tool ALPHA (SUT) is a command line (CLI) tool that checks a Magento instance against a specific version by analyzing all the non-Magento modules installed in it.
+
+The SUT identifies potential problems that must be fixed in your custom code before attempting to upgrade to a newer version of Magento. SUT complements another experimental tool called [ampersand-magento2-upgrade-patch-helper](https://github.com/AmpersandHQ/ampersand-magento2-upgrade-patch-helper).
```
Please remove the mention to ampersand "SUT complements another experimental tool called [ampersand-magento2-upgrade-patch-helper](https://github.com/AmpersandHQ/ampersand-magento2-upgrade-patch-helper)."

danidelcar replied:
done

sgonzalezmangana commented on src/safe-upgrade-tool/install.md:
```diff
@@ -0,0 +1,100 @@
+
+The following diagram shows the expected workflow when running the SUT:
+
+![SUT Diagram](img/mvp-diagram.png)
```
Please replace this image with this one:
![image](https://user-images.githubusercontent.com/10559423/105175070-bf83c900-5b23-11eb-9a23-a139e53211e8.png)

danidelcar replied:
added

sgonzalezmangana commented on src/safe-upgrade-tool/install.md:
```diff
@@ -0,0 +1,100 @@
+![SUT Diagram](img/mvp-diagram.png)
+
+### Who is the SUT for?
+
```
Let's add this here:
"The SUT Alpha is now available for all Magento Commerce merchants, and it can be downloaded on Magento repository [repo link]."

### SUT main use case

danidelcar replied:
added link

sgonzalezmangana commented on src/safe-upgrade-tool/install.md:
```diff
@@ -0,0 +1,100 @@
+### Who is the SUT for?
+
+The following use case describes the typical process for a Magento partner to upgrade a client's Magento instance:
+
```
Let's add this image here
![image](https://user-images.githubusercontent.com/10559423/105175833-bba47680-5b24-11eb-87b5-bd350f8efa41.png)

danidelcar replied:
added

sgonzalezmangana replied:
I can't see it but great that you add it. Thanks

sgonzalezmangana commented on src/safe-upgrade-tool/install.md:
```diff
@@ -0,0 +1,100 @@
+
+### Recommended actions
+
+It is recommended to avoid naming SUT modules with the same name, as a segmentation fault error can occur.
```
This is not a SUT issue it is a Magento limitation, the text should be changed for:
"Magento best practice is not to have 2 modules with the same name, if this happens SUT will show a segmentation fault error in which case you'll have to analyze each module independently with the option -m:"

@svera @jcuerdo @paliarush please validate

sgonzalezmangana commented on src/safe-upgrade-tool/install.md:
```diff
@@ -0,0 +1,100 @@
+
+It is recommended to avoid naming SUT modules with the same name, as a segmentation fault error can occur.
+
+If this segmentation error takes place, solve it by:
```
Please remove this:
"If this segmentation error takes place, solve it by:

*  Avoid analyzing a set of modules if there are 2 modules with the same name.
*  Analyze modules separately with the option -m:"

sgonzalezmangana commented on src/safe-upgrade-tool/install.md:
````diff
@@ -0,0 +1,100 @@
+    ```bash
+    bin/sut upgrade:check /(instance_path) --coming-version=2.4.1 -m /(module_path)
+    ```
+
````
Please add:
"If you get memory issues while executing SUT our recommendation is to use the -m command to run the tool against a specific module, this will prevent the problem to happen"

sgonzalezmangana commented on src/safe-upgrade-tool/install.md:
```diff
@@ -0,0 +1,100 @@
+
+### Magento access keys
+
+You must have [Magento access keys]({{site.baseurl}}/marketplace/sellers/profile-information.html#access-keys) to use the SUT. Add your Magento access keys to your `auth.json` file, which is located at `~/.composer` by default.
```
let's add download before use:
"to download and use the SUT."

sgonzalezmangana commented on src/safe-upgrade-tool/run.md:
```diff
@@ -0,0 +1,143 @@
+
+### Memory limitations
+
+If you are running a system with large third-party modules and files, the SUT requires a high amount of RAM, at least 2GB RAM.
```
This should be required on the prerequisites

sgonzalezmangana commented on src/safe-upgrade-tool/install.md:
```diff
@@ -0,0 +1,100 @@
+
+{:.bs-callout-info}
+You can run the SUT in any operating system. There is no requirement to run the SUT where your Magento instance is located. It is necessary for SUT to have access to the source code of the Magento instance. For example, you can install the SUT on one server and point it at your Magento installation on another server.
+
```
If you are running SUT against a Magento customized instance with large third-party modules and files, the tool might require a high amount of RAM, at least 2GB RAM.

sgonzalezmangana commented on src/safe-upgrade-tool/run.md:
```diff
@@ -0,0 +1,143 @@
+
+If you are running a system with large third-party modules and files, the SUT requires a high amount of RAM, at least 2GB RAM.
+
+### Executing the tool
```
On this area "Executing the tool" we must add the -m command as it was developed by the team to execute the tool against a specific directory/module

danidelcar commented:
running tests

danidelcar commented:
running tests

danidelcar commented:
running tests

danidelcar commented:
running tests
